### ISSUE TITLE
Issue with MasterSlaveConnection

### ISSUE THREAD

**Issue by gcavana** [jsor/doctrine-postgis#17]
Why do we need this for https://github.com/jsor/doctrine-postgis/blob/master/src/Event/DBALSchemaEventSubscriber.php#L62 ?

There is an issue with MasterSlaveConnection because you have two connections, master and slave.

**Comment by jsor** [comment]
This is because the subcriber needs a connection to work and should not allow to switch the connection it currently works on.

I've added a patch (a771ec7465f1849b8819f6ec331fd2e4df38c117) to allow multiple connections. Could you test the [multiple-connection-subscriber](https://github.com/jsor/doctrine-postgis/tree/multiple-connection-subscriber)  branch and let me know if it works for you?

**Comment by gcavana** [comment]
Well thank you for the quick answer i will try it tomorrow morning.

**Comment by NoiseByNorthwest** [comment]
@jsor This is more an issue about MasterSlaveConnection than multiple connections. In fact in our use case there is only one connection which is a MasterSlaveConnection. This kind of connection can enter a connection process up to 2 times (by default a slave connection is established and then a master if required later for write op). So IMHO the patch is simply to discard the exception in case if multi postprocess call on the same connection object, which can be filtered with

``` php
if ($this->postConnectCalled) {
    if ($this->connection === $args->getConnection()) {
        return;
    }
    ...
}
```

Furthermore, your multiple connection patch seems to be broken since you are invoking postConnect() (https://github.com/jsor/doctrine-postgis/blob/multiple-connection-subscriber/src/Event/DBALSchemaEventSubscriber.php#L67) on a subscriber with its postConnectCalled flag set to true (previously cloned from $this), this will explode the call stack.

**Comment by jsor** [comment]
@NoiseByNorthwest Thanks for the review, and you're totally right (has been this kind of late-night-after-a long-work-day patch where you try to be smart 😊)!

I've implemented your solution (f3bcb2688a66efe586da517faafca83a283be72d) in the [subscriber-multiple-connect](https://github.com/jsor/doctrine-postgis/tree/subscriber-multiple-connect) branch.

@gcavana Could you please test the `subscriber-multiple-connect` branch instead?

**Comment by gcavana** [comment]
@jsor ok thx !

**Comment by NoiseByNorthwest** [comment]
@jsor thx for your patch ! We've just test it and all is ok.

**Comment by jsor** [comment]
Fix released in [v1.1.1](https://github.com/jsor/doctrine-postgis/releases/tag/v1.1.1). Thank you to you both!
